zcash_primitives/src/transaction/components/sapling.rs

```diff
@@ -694,45 +163,34 @@ impl OutputDescription<GrothProofBytes> {
 
         let zkproof = read_zkproof(&mut reader)?;
 
-        Ok(OutputDescription {
+        Ok(OutputDescription::from_parts(
             cv,
             cmu,
             ephemeral_key,
             enc_ciphertext,
             out_ciphertext,
             zkproof,
-        })
+        ))
     }
 
     pub fn write_v4<W: Write>(&self, mut writer: W) -> io::Result<()> {
-        writer.write_all(&self.cv.to_bytes())?;
-        writer.write_all(self.cmu.to_bytes().as_ref())?;
-        writer.write_all(self.ephemeral_key.as_ref())?;
-        writer.write_all(&self.enc_ciphertext)?;
-        writer.write_all(&self.out_ciphertext)?;
-        writer.write_all(&self.zkproof)
+        writer.write_all(&self.cv().to_bytes())?;
+        writer.write_all(self.cmu().to_bytes().as_ref())?;
+        writer.write_all(self.ephemeral_key().as_ref())?;
+        writer.write_all(self.enc_ciphertext())?;
+        writer.write_all(self.out_ciphertext())?;
+        writer.write_all(self.zkproof())
     }
 
     pub fn write_v5_without_proof<W: Write>(&self, mut writer: W) -> io::Result<()> {
-        writer.write_all(&self.cv.to_bytes())?;
-        writer.write_all(self.cmu.to_bytes().as_ref())?;
-        writer.write_all(self.ephemeral_key.as_ref())?;
-        writer.write_all(&self.enc_ciphertext)?;
-        writer.write_all(&self.out_ciphertext)
+        writer.write_all(&self.cv().to_bytes())?;
+        writer.write_all(self.cmu().to_bytes().as_ref())?;
+        writer.write_all(self.ephemeral_key().as_ref())?;
+        writer.write_all(self.enc_ciphertext())?;
+        writer.write_all(self.out_ciphertext())
     }
 }
 
-#[derive(Clone)]
-pub struct OutputDescriptionV5 {
-    cv: ValueCommitment,
-    cmu: ExtractedNoteCommitment,
-    ephemeral_key: EphemeralKeyBytes,
-    enc_ciphertext: [u8; 580],
-    out_ciphertext: [u8; 80],
-}
-
-memuse::impl_no_dynamic_usage!(OutputDescriptionV5);
-
 impl OutputDescriptionV5 {
```

Comment:
Same comment with this impl.

Reply:
I was planning to change these in a separate PR, to keep the diff on this PR smaller.